### PR TITLE
fix: white image stuck in onboarding

### LINKED ISSUE #4629: [Onboarding scene] White screen on start
The onboarding scene is showing a white screen on start for a split of a second. 

Looks like the correct image [is loading late](https://decentralandteam.slack.com/archives/C01QVR7TJK1/p1678708121540229?thread_ts=1678705654.506769&cid=C01QVR7TJK1), and what we see is an empty raw image that is not ready to be shown

### PR DESCRIPTION
**Pull request by dalkia**
## What does this PR change?

Fixes #4629

To white image issue came from enabling a texture before it has been fetched. This PR reorganizes the call

The stuck situation came from changing realms at start, and generated race conditions with scenes not being unloaded properly. This PR takes you directly to the onboarding realm if the flag is on 

Also cleans old loading screen remanant code

## How to test the changes?

1. Go to zone (only place where flag is enabled) in incognito window. 
2. Create a user
3. Teleport out of the onboarding (you can use the genesis plaza button, or complete the onboarding)
4. Get to Genesis Plaza
5. Refresh
6. Check that the onboarding did not start again
7. With another already created account, check that you are not being taken to the onboarding

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

File: browser-interface/packages/shared/dao/sagas.ts [hunk 1]
```diff
@@ -153,13 +167,44 @@ function* selectRealm() {
...
 }
 
+function OnboardingTutorialRealm(){
+  const profile = getCurrentUserProfile(store.getState())!
+  if(profile){
+    const NEEDS_TUTORIAL = RESET_TUTORIAL || !profile.tutorialStep
+    const NEW_TUTORIAL_FEATURE_FLAG = getFeatureFlagVariantName(store.getState(), 'new_tutorial_variant')
+    const IS_NEW_TUTORIAL_DISABLED =
+      NEW_TUTORIAL_FEATURE_FLAG === 'disabled' || NEW_TUTORIAL_FEATURE_FLAG === 'undefined' || HAS_INITIAL_POSITION_MARK
+      console.log("DEBUG" + profile.tutorialStep);
```
Comment (Kinerius) [inline browser-interface/packages/shared/dao/sagas.ts:185]: You probably want to remove this :D

**Review by Kinerius — APPROVED** [review]
LGTM!

File: browser-interface/packages/shared/dao/sagas.ts [hunk 2]
```diff
@@ -160,6 +175,35 @@ function* selectRealm() {
   return realm
 }
 
+function* onboardingTutorialRealm() {
+  const profile = yield select(getCurrentUserProfile)
+  if (profile) {
+    const needs_tutorial = RESET_TUTORIAL || !profile.tutorialStep
+    const new_tutorial_feature_flag = getFeatureFlagVariantName(store.getState(), 'new_tutorial_variant')
```
Comment (menduz) [inline browser-interface/packages/shared/dao/sagas.ts:182]: Style and conventions:
* use camel case
* use yield select inside sagas
```suggestion
    const newTutorialFeatureFlag = yield select(getFeatureFlagVariantName, 'new_tutorial_variant')
```

Comment (dalkia) [reply browser-interface/packages/shared/dao/sagas.ts:182]: Correcting. Thanks!

File: browser-interface/packages/shared/dao/sagas.ts (hunk 1 above)
Comment (dalkia) [reply browser-interface/packages/shared/dao/sagas.ts]: Removed!

File: browser-interface/packages/shared/dao/sagas.ts [hunk 3]
```diff
@@ -160,6 +177,35 @@ function* selectRealm() {
...
+  const profile = yield select(getCurrentUserProfile)
+  if (profile) {
+    const needsTutorial = RESET_TUTORIAL || !profile.tutorialStep
+    const newTutorialFeatureFlag = yield select(getFeatureFlagVariantName, 'new_tutorial_variant')
+    const isNewTutorialDisabled =
+      newTutorialFeatureFlag === 'disabled' || newTutorialFeatureFlag === 'undefined' || HAS_INITIAL_POSITION_MARK
+    console.log('DEBUG' + profile.tutorialStep)
+    if (needsTutorial && !isNewTutorialDisabled) {
+      try {
+        const realm: string | undefined = getFeatureFlagVariantValue(store.getState(), 'new_tutorial_variant')
```
Comment (menduz) [inline browser-interface/packages/shared/dao/sagas.ts:190]:
```suggestion
        const realm: string | undefined = yield select(getFeatureFlagVariantValue, 'new_tutorial_variant')
```

File: browser-interface/packages/shared/dao/sagas.ts [hunk 4]
```diff
@@ -216,6 +262,15 @@ function* cacheCatalystRealm() {
   const dao: string = yield select((state) => state.dao)
   const realmAdapter: IRealmAdapter = yield call(waitForRealm)
 
+  if (
+    isOnboarding &&
+    realmAdapter.about.configurations?.realmName !==
+      getFeatureFlagVariantValue(store.getState(), 'new_tutorial_variant')
+  ) {
+    store.dispatch(saveProfileDelta({ tutorialStep: 256 }))
```
Comment (menduz) [inline browser-interface/packages/shared/dao/sagas.ts:270]: using store.dispatch inside a saga is forbidden and breaks the code, use `yield put(saveProfileDelta({ tutorialStep: 256 }))` instead
```suggestion
    yield put(saveProfileDelta({ tutorialStep: 256 }))
```

File: browser-interface/packages/shared/dao/sagas.ts [hunk 5]
```diff
@@ -216,6 +262,15 @@ function* cacheCatalystRealm() {
   const dao: string = yield select((state) => state.dao)
   const realmAdapter: IRealmAdapter = yield call(waitForRealm)
 
+  if (
+    isOnboarding &&
+    realmAdapter.about.configurations?.realmName !==
+      getFeatureFlagVariantValue(store.getState(), 'new_tutorial_variant')
```
Comment (menduz) [inline browser-interface/packages/shared/dao/sagas.ts:268]: * where does `isOnboarding` comes from?
* can you unwind this if into `if (two && consts)`?

Comment (dalkia) [reply browser-interface/packages/shared/dao/sagas.ts:268]: isOnboarding is a bool that determines we are doing the onboarding. I needed some way to detect that we were inOboarding when a realmChanged happen, so as to set the onboarding as complete. I could as well have started another async function that checks that, but not completely sure on how to do it.
That would have saved me the isOnboarding var

File: browser-interface/packages/shared/dao/sagas.ts [hunk 6]
```diff
@@ -160,6 +176,34 @@ function* selectRealm() {
...
+function* onboardingTutorialRealm() {
+  const profile = yield select(getCurrentUserProfile)
+  if (profile) {
+    const needsTutorial = RESET_TUTORIAL || !profile.tutorialStep
+    const newTutorialFeatureFlag = yield select(getFeatureFlagVariantName, 'new_tutorial_variant')
+    const isNewTutorialDisabled =
+      newTutorialFeatureFlag === 'disabled' || newTutorialFeatureFlag === 'undefined' || HAS_INITIAL_POSITION_MARK
+    if (needsTutorial && !isNewTutorialDisabled) {
+      try {
+        const realm: string | undefined = yield select(getFeatureFlagVariantValue, 'new_tutorial_variant')
```
Comment (eordano) [inline browser-interface/packages/shared/dao/sagas.ts:188]: Isn't this the same as `newTutorialFeatureFlag` of line 183? Avoid doing multiple consecutive select calls if possible.

**Review by eordano — COMMENTED** [review]
Code looks good, and should work. Tested that it works correctly.

I do thing though that we should strip out the "read catalyst from localstorage" part of the code. It's adding a lot of unnecessary complexity...

File: browser-interface/packages/shared/dao/sagas.ts [hunk 7]
```diff
@@ -35,8 +44,14 @@ import { createAlgorithm } from './pick-realm-algorithm/index'
...
 import { Candidate, PingResult, Realm, ServerConnectionStatus } from './types'
 import { ask, ping } from './utils/ping'
+import { trackEvent } from '../analytics/trackEvent'
+import { logger } from '../../entryPoints/logger'
+import { getCurrentUserProfile } from '../profiles/selectors'
+import { saveProfileDelta } from '../profiles/actions'
 
 const waitForExplorerIdentity = waitFor(getCurrentIdentity, USER_AUTHENTICATED)
+let isCurrentlyInOnboarding = false
+let onboardingRealmName = ""
```
Comment (menduz) [inline browser-interface/packages/shared/dao/sagas.ts:54]: prevent using global variables, use selector instead

**Pull request merged**